### PR TITLE
PAYARA-3992 JSON-B Serialisation Error With Custom Serialiser

### PR DESCRIPTION
**Pull request by MattGill98**
I've given the user serialiser a fallback serialiser, so that instead of throwing an exception it falls back to that. Seems to work okay for recursive objects as well as the customer use case.

### CODE REVIEW

**Comment by MattGill98** [comment]
Sorry for the mess, had an absolute mare with GPG keys

File: src/main/java/org/eclipse/yasson/internal/model/PropertyModel.java
Comment (pdudits) [inline src/main/java/org/eclipse/yasson/internal/model/PropertyModel.java:3]: I believe the correct copyright is adding a line
```
Copyright (c) 2019 Payara Foundation and/or its affiliates. All rights reserved
```

**Review by pdudits — CHANGES_REQUESTED** [review]
I like how this works, but we're going to territory which is not covered by the spec. 

So we'd need to check if there is at least de-facto agreement on how this should work (i. e. by checking what johnzon does), and proposing a spec change. In the meantime, this should be flagged as experimental/vendor feature i. e. by custom property in the config.

**Review by jGauravGupta — APPROVED** [review]

File: src/main/java/org/eclipse/yasson/internal/model/PropertyModel.java
Comment (Pandrex247) [reply src/main/java/org/eclipse/yasson/internal/model/PropertyModel.java:3]: I'm not convinced myself, I think the way Matt has done it is correct.
Note: I'm not a copyright expert.

Comment (Pandrex247) [reply src/main/java/org/eclipse/yasson/internal/model/PropertyModel.java:3]: Well, barring the fact it should be Payara Foundation rather than Payara Services

Comment (MattGill98) [reply src/main/java/org/eclipse/yasson/internal/model/PropertyModel.java]: I don't know what you're talking about 🧐

**Comment by MattGill98** [comment]
@pdudits as soon as this has your approval I'll merge both this and the main PR!

**Comment by Pandrex247** [comment]
@MattGill98 not without updating the patched project PR please :)

**Review by pdudits — APPROVED** [review]

**Pull request merged**
